docs/typescript.md

```diff
@@ -1,14 +1,209 @@
-# TypeScript Guidelines
+# Typescript Guidelines
 
-These guidelines specifically apply to TypeScript.
+**_Purpose_**: Establish best practices for writing robust, maintainable, and efficient TypeScript code.
 
-## Provide explicit return types for functions/methods
+**_Audience_**: Internal and external developers using TypeScript in Metamask.
+
+## **Type**
+
+By defining expected object structures with interfaces, TypeScript promotes developer intent and unlocks powerful IDE features like accurate completion and context-aware navigation.
+
+### **Enums**
+
+TypeScript offers several tools for crafting clear data definitions, with enumerations and unions standing as popular choices.
+
+**Consider using enums over union types for situations with a fixed set of known values.**
```

Comment:
The entries in this doc should be at the same heading level.

```suggestion
#### Consider using enums over union types for situations with a fixed set of known values.
```

Reply:
I'm using `**` to represent lower level title, but can certainly change everyone !